From e0b11514b8dab7b38b428833c8ef4a9f32ce72a6 Mon Sep 17 00:00:00 2001
Message-Id: <patch-v2-06.10-e0b11514b8d-20211022T192811Z-avarab@gmail.com>
In-Reply-To: <cover-v2-00.10-00000000000-20211022T192811Z-avarab@gmail.com>
References: <cover-0.8-00000000000-20211020T183533Z-avarab@gmail.com>
	<cover-v2-00.10-00000000000-20211022T192811Z-avarab@gmail.com>
From: Johannes Sixt <j6t@kdbg.org>
Date: Wed, 4 Jul 2018 09:15:34 +0200
Subject: [PATCH v2 06/10] generate-cmdlist.sh: replace for loop by printf's
 auto-repeat feature
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This is just a small code reduction. There is a small probability that
the new code breaks when the category list is empty. But that would be
noticed during the compile step.

Signed-off-by: Johannes Sixt <j6t@kdbg.org>
Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 generate-cmdlist.sh | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/generate-cmdlist.sh b/generate-cmdlist.sh
index e517c33710a..a1ab2b1f077 100755
--- a/generate-cmdlist.sh
+++ b/generate-cmdlist.sh
@@ -67,10 +67,7 @@ print_command_list () {
 	while read cmd rest
 	do
 		printf "	{ \"$cmd\", $(get_synopsis $cmd), 0"
-		for cat in $(echo "$rest" | get_category_line)
-		do
-			printf " | CAT_$cat"
-		done
+		printf " | CAT_%s" $(echo "$rest" | get_category_line)
 		echo " },"
 	done
 	echo "};"
-- 
2.33.1.1505.g075a284c562

